File: algorithms/majority_element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  let count = 1;
  let ans = nums[0];

  for (let i = 1; i < nums.length; i++) {
    if (count === 0) ans = nums[i];
    ans === nums[i] ? count++ : count--;
  }

  return ans;
};

File: swift/MajorityElement.swift
class majorityElement {
    func majorityElement(_ nums: [Int]) -> Int {
        var candidate = 0
        var count = 0

        for num in nums {
            if count == 0 {
                candidate = num
                count += 1
            } else if candidate == num {
                count += 1
            } else {
                count -= 1
            }
        }

        return candidate
    }
}